src/Compilers/CSharp/Portable/Lowering/MethodToClassRewriter.cs

```diff
@@ -368,7 +364,12 @@ public override BoundNode VisitAwaitExpression(BoundAwaitExpression node)
         {
             BoundExpression expression = (BoundExpression)this.Visit(node.Expression);
             TypeSymbol type = this.VisitType(node.Type);
-            return node.Update(expression, VisitMethodSymbol(node.GetAwaiter), VisitPropertySymbol(node.IsCompleted), VisitMethodSymbol(node.GetResult), type);
+
+            AwaitableInfo info = node.AwaitableInfo;
+            return node.Update(
+                expression,
+                new AwaitableInfo(VisitMethodSymbol(info.GetAwaiter), VisitPropertySymbol(info.IsCompleted), VisitMethodSymbol(info.GetResult)),
+                type);
```

Comment:
The `new AwaitableInfo()` will force update of `node` and all parent nodes. If that's not expected, perhaps add an `AwaitableInfo.Update` method.